tests/testthat/test-calculate_stay.R

```diff
@@ -0,0 +1,172 @@
+test_that("Can return correct length of stay", {
+
+  ## Single calculations
+
+  # Normal calculation - no sc_qtr supplied
+  # if start and end dates are both supplied, calculate length of stay
+
+  # Start before FY, end during FY
+  expect_equal(calculate_stay("1920", as.Date("2019/03/31"), as.Date("2019/10/31")), 214)
+  # Start during FY, end during FY
+  expect_equal(calculate_stay("1920", as.Date("2019/06/30"), as.Date("2019/08/31")), 62)
+  # Start before FY, end after FY
+  expect_equal(calculate_stay("1920", as.Date("2019/01/01"), as.Date("2020/04/01")), 456)
+  # Start during FY, end after FY
+  expect_equal(calculate_stay("1920", as.Date("2019/04/01"), as.Date("2020/07/01")), 457)
+
+  # Normal calculation - no sc_qtr supplied
+  # if start date supplied but end date missing, use dummy date (end_FY + 1) to calculate length of stay
+
+  # Start before FY, end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/03/31"), as.Date(NA)), 367)
+  # Start during FY, end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/06/30"), as.Date(NA)), 276)
+  # Start before FY, end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/01/01"), as.Date(NA)), 456)
+  # Start during FY, end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/04/01"), as.Date(NA)), 366)
+
+
+  # SC calculation - sc_qtr supplied
+  # if start date supplied but end date missing, use end_qtr to calculate length of stay
+
+  # Start before FY, end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/03/31"), as.Date(NA), "2019Q1"), 92)
+  # Start during FY, end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/06/30"), as.Date(NA), "2019Q2"), 93)
+  # Start before FY, end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/01/01"), as.Date(NA), "2019Q3"), 365)
+  # Start during FY, end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/04/01"), as.Date(NA), "2019Q4"), 366)
+
+  # SC calculation - sc_qtr supplied
+  # if qtr_end < start_date then set to next qtr to calculate length of stay
+
+  # qtr_end < start_date , end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/07/31"), as.Date(NA), "2019Q1"), 62)
+  # qtr_end < start_date , end missing
+  expect_equal(calculate_stay("1920", as.Date("2019/10/31"), as.Date(NA), "2019Q2"), 62)
+  # qtr_end < start_date , end missing
+  expect_equal(calculate_stay("1920", as.Date("2020/01/31"), as.Date(NA), "2019Q3"), 61)
+  # qtr_end < start_date , end missing
+  expect_equal(calculate_stay("1920", as.Date("2020/04/30"), as.Date(NA), "2019Q4"), 62)
+
+
+
+  ## Vectors
+
```

Comment:
Can you remove some whitespace here please, and create a new testthat section too `test_that("Can calculate the correct stay for vectors of dates", {..})`